src/coreclr/jit/gentree.cpp

```diff
@@ -25728,18 +26537,6 @@ GenTree* Compiler::gtNewSimdShuffleNode(
     uint64_t value  = 0;
     simd_t   vecCns = {};
 
-    if (simdSize == 16)
-    {
-        // Vector128.Shuffle(a, Vector128.Create(2, 3, 0, 1)) -> ExtractVector128(v.AsUInt64(), v.AsUInt64(), 1)
-        if ((op2->GetIntegralVectorConstElement(0, TYP_ULONG) == 0x300000002) &&
-            (op2->GetIntegralVectorConstElement(1, TYP_ULONG) == 0x100000000))
-        {
-            GenTree* op1Clone = fgMakeMultiUse(&op1);
-            return gtNewSimdHWIntrinsicNode(type, op1, op1Clone, gtNewIconNode(1), NI_AdvSimd_ExtractVector128,
-                                            CORINFO_TYPE_ULONG, simdSize);
-        }
-    }
-
```

Comment:
@tannergooding for when you're reviewing: I believe the code here before was incorrect, as it didn't check the element size (and thus I think it could fire for something like `V128.Shuffle(a, V128.Create((byte)2, 0, 0, 0, 3, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0))`). I've moved it later and checked `vecCns` instead, this way it should work for all cases of lane swapping. Please let me know if I'm missing something and should revert it.